File: tests/unreached/extension.swift
extension Int {
  var double: Int { return self * 2 }
  func square() -> Int {
    return (self * self);
  }
}

extension Double {
  var double: Double { return self * 2 }
}

var a:Int = 8.double.square();
var oneInch = 25.75.double;

expect(a == 256);
expect(oneInch == 51.5);

File: src/Parser/declare/extension.js
import {
  Token,
  Types as Type,
  TokenList as TT
} from "../../labels";

import Node from "../../nodes";

/**
 * @return {Node}
 */
export function parseExtension() {

  let node = new Node.ExtensionDeclaration();

  this.expect(TT.EXTENSION);

  node.argument = this.parseLiteral();

  if (this.eat(TT.COLON)) {
    node.extend = this.parseTypeInheritance();
  }

  if (this.eat(TT.LBRACE)) {
    node.body = this.parseBlock();
    this.expect(TT.RBRACE);
  }

  return (node);

}